src/intpack.f90

```diff
@@ -1476,43 +1472,48 @@ double precision function aainer(l,m,n,d,dp,fnu,fn,fd)
          fijk=fn(knt)*fd(kn)*fd(k+1)*dp(ijkt+1)*fij
          lmrsij=lmnrst-2*ijkt
          last4 = ilrh+1
-         do 3 iu=1,last4
+         do iu=1,last4
           u = iu-1
           ilru=ilr-2*u
 !sg   if (u+1.gt.9 .or. ilru.gt.9) stop 'aainer: ilru'
           fu=fd(u+1)*fd(ilru)
-          if(dabs(d(1))-0.0000001d0)10,10,11
-   10     if(ilru-1) 12,12, 3
-   11     fu=fu*d(1)**(ilru-1)
-   12     last5 = jmsh+1
-          do 2 iv=1,last5
+          if(dabs(d(1))-0.0000001d0.le.0.0) then
+           if(ilru-1.gt.0) cycle
+          else
+           fu=fu*d(1)**(ilru-1)
+          end if
+          last5 = jmsh+1
+          do iv=1,last5
            v = iv-1
            jmsv=jms-2*v
            fuv=fu*fd(v+1)*fd(jmsv)
-           if(dabs(d(2))-0.0000001d0)20,20,21
-   20      if(jmsv-1) 22,22,2
-   21      fuv=fuv*d(2)**(jmsv-1)
-   22      last6 = knth+1
-           do 1 iw=1,last6
+           if(dabs(d(2))-0.0000001d0.le.0d0) then
+            if(jmsv-1.gt.0) cycle
+           else
+            fuv=fuv*d(2)**(jmsv-1)
+           end if
+           last6 = knth+1
+           do iw=1,last6
             w = iw-1
             kntw=knt-2*w
             fuvw=fuv*fd(w+1) *fd(kntw)
-            if(dabs(d(3))-0.0000001d0)30,30,31
-   30       if(kntw-1) 32,32,1
-   31       fuvw=fuvw*d(3)**(kntw-1)
-   32       uvwt=u+v+w
+            if(dabs(d(3))-0.0000001d0.le.0d0) then
+             if(kntw-1.gt.0) cycle
+            else
+             fuvw=fuvw*d(3)**(kntw-1)
+            end if
+            uvwt=u+v+w
```

Comment:
Second arithmetic if leads to end of loop with label `1` (-> `cycle`). Refactor captures logic correctly.